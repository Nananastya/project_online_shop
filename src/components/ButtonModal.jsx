import React from 'react';
import Button from 'react-bootstrap/Button';

export default function ButtonModal(props) {
  return (
    <Button variant="primary" 
      onClick={(e) => {
        props.onClick(e);
      }}
      style = {props.backgroundColor}>
      {props.text}
    </Button>
  )
}