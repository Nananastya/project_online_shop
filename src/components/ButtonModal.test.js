import ButtonModal from "./ButtonModal";
import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

test("test button", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <ButtonModal
        onClick={handleClick}
        text="Click me"
      />
    );
  
    const button = screen.getByText("Click me");
    fireEvent.click(button);
  
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
