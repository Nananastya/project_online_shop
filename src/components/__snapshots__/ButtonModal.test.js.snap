// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button 1`] = `
<div>
  <button
    class="btn btn-primary"
    type="button"
  >
    Click me
  </button>
</div>
`;
